Makefile

```diff
@@ -51,9 +52,15 @@ build: setup
 build-windows: setup
 	CGO_ENABLED=0 GOOS=windows go build -a -ldflags ${LDFLAGS} -o _output/secrets-store-csi.exe ./cmd/secrets-store-csi-driver
 image:
-	docker buildx build --no-cache --build-arg LDFLAGS=${LDFLAGS} -t $(IMAGE_TAG) -f Dockerfile --platform="linux/amd64" --output "type=docker,push=false" .
+	docker buildx build --no-cache --build-arg LDFLAGS=${LDFLAGS} -t $(IMAGE_TAG) -f docker/Dockerfile --platform="linux/amd64" --output "type=docker,push=false" .
```

Comment:
there is no `docker/` dir?

Reply:
Indeed, I moved some things on rebase (there was a merge conflict). Because of some recent changes, I had to rewrite some bits of the building process, and I moved everything in the root directory.

I still prefer having the docker folder. The reason is because we're configuring the Image Promoter to build the image when changes to files in the ``docker/.*`` are being made (for example, you'd trigger the builder on an image version bump). Having everything on the root directory would mean that the builder would be triggered on every PR, which is unnecessary.

Reply:
@claudiubelu I'm just wondering if we can make it more architecture natural so that we can build other platforms as well like arm, ppc64le and s390x as well?

Reply:
I did, in the ``build.sh`` below.

As I mentioned before, I've made a separate ``docker/Makefile`` which is going to be used by the Image Builder / Promoter. This target is still being used for testing, and is called in this ``Makefile``. This only builds the ``linux/amd64`` image, which is sufficient for testing, while mine builds all images, which takes a lot more time, especially due to the Windows images.

Reply:
cool